src/libdredd/src/mutation_replace_binary_operator.cc

```diff
@@ -1024,34 +1027,81 @@ MutationReplaceBinaryOperator::ClangOperatorKindToProtobufOperatorKind(
 
 bool MutationReplaceBinaryOperator::
     IsRedundantReplacementForBooleanValuedOperator(
-        clang::BinaryOperatorKind operator_kind) const {
+        clang::BinaryOperatorKind replacement_operator,
+        const clang::ASTContext& ast_context) const {
+  if (binary_operator_->getLHS()->getType()->isUnsignedIntegerType() &&
+      binary_operator_->getRHS()->getType()->isUnsignedIntegerType()) {
+    return IsRedundantReplacementForUnsignedComparison(replacement_operator,
+                                                       ast_context);
+  }
+
   switch (binary_operator_->getOpcode()) {
     // From
     // https://people.cs.umass.edu/~rjust/publ/non_redundant_mutants_jstvr_2014.pdf:
     // For boolean operators, only a subset of replacements are non-redundant.
     case clang::BO_LAnd:
-      return operator_kind != clang::BO_EQ;
+      return replacement_operator != clang::BO_EQ;
     case clang::BO_LOr:
-      return operator_kind != clang::BO_NE;
+      return replacement_operator != clang::BO_NE;
     case clang::BO_GT:
-      return operator_kind != clang::BO_GE && operator_kind != clang::BO_NE;
+      return replacement_operator != clang::BO_GE &&
+             replacement_operator != clang::BO_NE;
     case clang::BO_GE:
-      return operator_kind != clang::BO_GT && operator_kind != clang::BO_EQ;
+      return replacement_operator != clang::BO_GT &&
+             replacement_operator != clang::BO_EQ;
     case clang::BO_LT:
-      return operator_kind != clang::BO_LE && operator_kind != clang::BO_NE;
+      return replacement_operator != clang::BO_LE &&
+             replacement_operator != clang::BO_NE;
     case clang::BO_LE:
-      return operator_kind != clang::BO_LT && operator_kind != clang::BO_EQ;
+      return replacement_operator != clang::BO_LT &&
+             replacement_operator != clang::BO_EQ;
     case clang::BO_EQ:
-      return operator_kind != clang::BO_LE && operator_kind != clang::BO_GE;
+      return replacement_operator != clang::BO_LE &&
+             replacement_operator != clang::BO_GE;
     case clang::BO_NE:
-      return operator_kind != clang::BO_LT && operator_kind != clang::BO_GT;
+      return replacement_operator != clang::BO_LT &&
+             replacement_operator != clang::BO_GT;
+    default:
+      return false;
+  }
+}
+
+bool MutationReplaceBinaryOperator::IsRedundantReplacementForUnsignedComparison(
+    clang::BinaryOperatorKind replacement_operator,
+    const clang::ASTContext& ast_context) const {
+  const bool valid_replacement = replacement_operator == clang::BO_NE ||
```

Comment:
I don't like the name `valid_replacement` as it suggests that if the replacement operator is anything else then the replacement is not valid, whereas this is only used under a given context.

I suggest naming it something mundane like `replacement_operator_is_eq_or_ne`.